### PR TITLE
[CI] Move appveyor tests to action

### PR DESCRIPTION
**Pull request by trivialfis**
* Drop support for VS14, use VS15 instead.
* Drop support for mingw.
* Remove debug build.
* Split up jvm tests.
* Split up Python tests.

### CODE REVIEW

**Review by RAMitchell — COMMENTED** [review]
So this retains mingw support?

**Review by RAMitchell — APPROVED** [review]

**Comment by hcho3** [comment]
Just removed the webhook for AppVeyor. Can you try merging in the `master` branch and try again?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -101,6 +101,66 @@ jobs:
...
 
+  python-tests:
+    name: Test XGBoost Python package
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        config:
+          - {os: windows-2016, compiler: 'mingw', python-version: "3.8"}
+          - {os: windows-2016, compiler: 'msvc', python-version: "3.8"}
+        python-version: ["3.8"]
```
Comment (hcho3) [inline .github/workflows/main.yml:112]: I don't see this test running in the status page?

**Comment by hcho3** [comment]
Can we add back MinGW? Also, we may want to install MinGW first. See https://github.com/marketplace/actions/install-mingw

**Comment by trivialfis** [comment]
Looked into the action you recommended, trying it now.

**Comment by trivialfis** [comment]
Nope, doesn't work.

**Comment by hcho3** [comment]
Not sure how well we can support MinGW. We should maybe drop it?

**Comment by trivialfis** [comment]
> Not sure how well we can support MinGW. We should maybe drop it?

So far I think the biggest consumer for mingw is R.  Personally I'm happy to drop support for mingw in non-R environment.  Nowadays VC is pretty good, also clang is easy to use on Windows ..

**Comment by trivialfis** [comment]
But since I'm not a Windows user so I can't speak for the others.

**Comment by hcho3** [comment]
@trivialfis Have you tried installing R in GitHub Actions? That should give us access to working MinGW.

File: .github/workflows/python_tests.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    - name: Build XGBoost with mingw
+      shell: bash -l {0}
+      if: matrix.config.compiler == 'mingw'
+      run: |
+        mkdir build_mingw
+        cd build_mingw
+        cmake ../ -G"Unix Makefiles" -DGOOGLE_TEST=ON  -DUSE_DMLC_GTEST=ON
+        make -j$(nproc)
+        ctest --extra-verbose
```
Comment (hcho3) [inline .github/workflows/python_tests.yml:79]: Please remove this section, since we won't test MinGW.

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: .github/workflows/python_tests.yml (hunk 2 above)
Comment (trivialfis) [reply .github/workflows/python_tests.yml]: Removed.

**Review by hcho3 — APPROVED** [review]

**Pull request merged**